plugin/store/store.go

```diff
@@ -98,12 +98,31 @@ func (ps *Store) SetState(p *v2.Plugin, state bool) {
 }
 
 // Add adds a plugin to memory and plugindb.
-func (ps *Store) Add(p *v2.Plugin) {
+// An error will be returned if there is a collision.
```

Comment:
@yongtang : Can you use `GetByName` in `Pull` to check for duplicate names and fail fast in the function? Then you can update `Add` to only check for collision by ID.